File: carRentalFrontend/src/app/module/customer/services/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../services/storage/storage.service';

const BASIC_URL="http://localhost:8080";

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http:HttpClient) { }

  getAllCars():Observable<any>{
    return this.http.get(BASIC_URL+"/api/customer/cars",{
      headers: this.createAuthorizationHeader()
    })
  }

  createAuthorizationHeader():HttpHeaders{
    let authHeader:HttpHeaders=new HttpHeaders();
    return authHeader.set('Authorization','Bearer '+StorageService.getToken())
  }

  getCarById(carId:number):Observable<any>{
    return this.http.get(BASIC_URL+"/api/customer/car/"+carId,{
      headers:this.createAuthorizationHeader()
    })
  }

  bookACar(bookACarDto: any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/customer/car/book",bookACarDto,{
      headers: this.createAuthorizationHeader()
    })
  }

  getBookingsByUserId():Observable<any>{
    return this.http.get(BASIC_URL+"/api/customer/car/bookings/"+StorageService.getUserId(),{
      headers:this.createAuthorizationHeader()
    })
  }

  searchCar(searchCarDto:any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/customer/car/search",searchCarDto,{
      headers:this.createAuthorizationHeader()
    });
  }

}

File: carRentalBackend/src/main/java/com/shubhi/service/customer/CustomerService.java
package com.shubhi.service.customer;

import java.util.List;

import com.shubhi.dto.BookACarDto;
import com.shubhi.dto.CarDto;
import com.shubhi.dto.CarDtoListDto;
import com.shubhi.dto.SearchCarDto;

import io.jsonwebtoken.io.IOException;

public interface CustomerService {
	List<CarDto> getAllCars();
	boolean bookACar(BookACarDto bookACarDto) throws IOException;
	CarDto getCarById(long carId);
	List<BookACarDto> getBookingsByUserId(long userId);
	CarDtoListDto searchCar(SearchCarDto searchCarDto);

}
